backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -532,30 +613,89 @@ public FutureExperimentRunDAO(
-                        .addBind(q -> q.bindList("authz_project_ids", projIds)),
-                executor);
+    queryFilterContextList.add(futureLocalContext);
+
+    // futurePredicatesContext
+    queryFilterContextList.add(
+        predicatesHandler.processPredicates(request.getPredicatesList(), executor));
+
+    // futureSortingContext
+    queryFilterContextList.add(
+        sortingHandler.processSort(request.getSortKey(), request.getAscending()));
+
+    // futureProjectIds based on workspace
+    if (request.getWorkspaceName().isEmpty()) {
+      queryFilterContextList.add(
+          getAllowedProjects(ModelDBActionEnum.ModelDBServiceActions.READ)
+              .thenCompose(
+                  projectIds -> {
+                    if (projectIds.isEmpty()) {
+                      // TODO: This is not works, need to update it.
+                      /*return InternalFuture.failedStage(new PermissionDeniedException(
+                      "Accessible projects not found in workspace: "
+                              + request.getWorkspaceName()));*/
+                    }
+                    return InternalFuture.completedInternalFuture(projectIds);
+                  },
+                  executor)
+              .thenApply(
+                  projIds ->
+                      new QueryFilterContext()
+                          .addCondition("experiment_run.project_id in (<authz_project_ids>)")
+                          .addBind(q -> q.bindList("authz_project_ids", projIds)),
+                  executor));
+    } else {
+      var requestProjectIds = new ArrayList<String>();
+      if (!request.getProjectId().isEmpty()) {
+        requestProjectIds.add(request.getProjectId());
+      }
+
+      final var futureProjectIds =
+          FutureGrpc.ClientRequest(
+                  uac.getWorkspaceService()
+                      .getWorkspaceByName(
+                          GetWorkspaceByName.newBuilder()
+                              .setName(request.getWorkspaceName())
+                              .build()),
+                  executor)
+              .thenCompose(
+                  workspace ->
+                      getAllowedResourceItems(
+                              requestProjectIds,
+                              workspace.getId(),
+                              ModelDBResourceEnum.ModelDBServiceResourceTypes.PROJECT)
+                          .thenCompose(
+                              getResourcesResponseItems -> {
+                                if (getResourcesResponseItems.isEmpty()) {
+                                  // TODO: This is not works, need to update it.
+                                  /*return InternalFuture.failedStage(
+                                  new PermissionDeniedException(
+                                      "Accessible project not found in workspace: "
+                                          + request.getWorkspaceName()));*/
```

Comment:
Same question

Reply:
Fixed in PR https://github.com/VertaAI/modeldb/pull/2242#pullrequestreview-648893239